blog/docs/events/security-audit-2023.md

```diff
@@ -0,0 +1,25 @@
+
+During the audit, Ada Logics was in close communication with the Knative team and shared findings ad hoc as the audit progressed. The Knative team and Ada Logics collaborated on mitigation and fixes of the issues found. 
+
+The audit mainly focused on the three core Knative subprojects: Eventing, Serving and Pkg, with a minor focus on Knative Extensions, Knative Func and Security-Guard.
+
+Knative Extensions are optional plugins and extensions for different Knative use cases. For example, users can find different officially maintained integrations for Knative Eventing, and the Security-Guard project also resides in Knative Extensions. Many Knative Extensions subprojects are not fully mature yet, with many having Alpha and Beta release statuses. 
+
+Ada Logics found 16 security issues of which all except for one have been fixed with upstream patches. Three of the found issues are in third-party dependencies to Knative. One of the issues found is scored with high severity due to its potential impact (ADA-KNATIVE-23-7). However, an attacker would need to compromise the Knative user’s supply-chain, which is highly unlikely to occur in a real-world scenario because few Knative users consume Knative in a way that would enable the attack. In addition, the attacker would need to time their attack to a high degree. The issue has been fixed.
+
+One CVE was assigned during the audit for a vulnerability that could allow an attacker with already escalated privileges to cause further damage in the cluster. The attacker needs to first establish a position in a Knative pod, and from there, they could exploit the vulnerability and cause denial of service of the Knative autoscaling, thereby denying any autoscaling of Knative. The issue was assigned CVE-2023-48713 of Moderate severity and has been fixed in v1.10.5, v1.12.0 and v1.11.3.
+
+The auditors found that Knative does not include provenance with releases; Provenance is a critical component of complying with [SLSA](https://slsa.dev/)  and ensuring tamper resistance of release artifacts. Recently, the SLSA community released v1.9.0 of the [slsa-github-generator](https://github.com/slsa-framework/slsa-github-generator) which produces SLSA L3 compliant provenance. slsa-github-generator ensures tamper-resistance of artifacts by producing verifiable provenance, thereby mitigating a series of known supply-chain risks, many of which have been exploited in the wild in recent years. Ada Logics recommend that Knative switches its build to the slsa-github-generator to comply with SLSA L3.
```

Comment:
Feel free to include this, but Knative is using [Prow](https://github.com/kubernetes/test-infra/tree/master/prow) to build and test releases (our usage requires more resources than are available on GitHub Actions, including real Kubernetes clusters).  As such, the GitHub Actions SLSA generator is probably not a good fit.

We're also using [`ko`](https://github.com/ko-build/ko) to build these releases, which doesn't currently have signing support (see https://github.com/ko-build/ko/issues/357 and https://github.com/ko-build/ko/issues/603).

Reply:
Huh, it turns out that _Knative Serving_ does not have attestations, but most of the other releases do.  For example:

```
cosign download attestation $(curl -L https://github.com/knative/eventing/releases/download/knative-v1.12.1/eventing-core.yaml | grep image: | head -1 | cut -d: -f2-)
```

(Grab the first image from the Eventing release and run `cosign download attestation` on it...)

Reply:
(And append ` | jq -r '.payload' | base64 --decode | jq` to read the actual in-toto attestation, rather than the base64-encoded envelope)

Reply:
https://github.com/knative/infra/issues/287 for the attestations for serving.

Reply:
```suggestion
```

Reply:
I would prefer to change that paragraph to mention Knatives good work on generating SLSA-compliant provenance than removing it. I have changed the paragraph in https://github.com/knative/docs/pull/5775/commits/b93cf114252ba04282d8a238cec99513350f7979

Reply:
I have also changed the executive summary and SLSA sections of the report to reflect this.